### PR TITLE
Updated docs post v0.1.20 release

### PR DESCRIPTION
**Pull request by AverageMarcus**
What this PR does / why we need it:

Updates all docs references with new version numbers.

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged): Fixes #

**Additional context**
Add any other context for the reviewers

/hold for https://github.com/kubernetes/k8s.io/pull/5978

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Image Builder Releases
 
-The current release of Image Builder is [v0.1.19][] (September 13, 2023). The corresponding container image is `registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.19`.
+The current release of Image Builder is [v0.1.20][] (September 13, 2023). The corresponding container image is `registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.20`.
```
Comment (mboersma) [inline RELEASE.md:3]: We should update “September 13” to today’s date.

Comment (AverageMarcus) [reply RELEASE.md]: I forgot about those. Updated now.

File: docs/book/src/capi/releasing.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Image Builder Releases
 
-The current release of Image Builder is [v0.1.19][] (September 13, 2023). The corresponding container image is `registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.19`.
+The current release of Image Builder is [v0.1.20][] (September 13, 2023). The corresponding container image is `registry.k8s.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.20`.
```
[the remark this answers is not in the document]
Comment (AverageMarcus) [reply docs/book/src/capi/releasing.md]: I forgot about those. Updated now.

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

Thanks for doing the release @AverageMarcus!

**Pull request merged**